javascript/core/src/verbs/util/expressions.ts

```diff
@@ -34,8 +36,10 @@ export function compareAll(
 		// cannot be satisfied
 		const comparisons = criteria.map(filter => {
 			const { value, operator, type } = filter
-			const right =
+			let right =
 				type === FilterCompareType.Column ? d[`${value.toString()}`]! : value
+			if(operator as DateComparisonOperator === DateComparisonOperator.AfterToday || operator as DateComparisonOperator === DateComparisonOperator.BeforeToday)
```

Comment:
moving the `new Date()` in the compareDates function would get rid of this special case here and put it in the type-specific comparison spot